diff -rupN src/Makefile src.new/Makefile
--- src/Makefile	1970-01-01 00:00:00.000000000 +0000
+++ src.new/Makefile	2014-03-06 21:00:13.195839400 +0000
@@ -0,0 +1,53 @@
+PRL_ETH ?= prl_eth/pvmnet
+PRL_TG ?= prl_tg/Toolgate/Guest/Linux/prl_tg
+PRL_FS ?= prl_fs/SharedFolders/Guest/Linux/prl_fs
+PRL_FREEZE ?= prl_fs_freeze/Snapshot/Guest/Linux/prl_freeze
+
+KVER ?= $(shell uname -r)
+KERNEL_DIR ?= /lib/modules/$(KVER)/build
+# Get version from version.h in kernel source directory
+ifeq "$(wildcard $(KERNEL_DIR)/include/linux/version.h)" ""
+	VERSION_H_PATH ?= $(KERNEL_DIR)/include/generated/uapi/linux/version.h
+else
+	VERSION_H_PATH ?= $(KERNEL_DIR)/include/linux/version.h
+endif
+
+VERSION_CODE := $(shell awk '/\#define[ \t]+LINUX_VERSION_CODE[ \t]\
+	*/ { print $$3 }' $(VERSION_H_PATH) | tail -n 1)
+
+all: build cleanme
+
+.PHONY: all
+
+build:
+	cd ${PRL_ETH} && make
+	cd ${PRL_TG} && make
+	cp -f ${PRL_TG}/*.symvers ${PRL_FS} ||:
+	cd ${PRL_FS} && make
+ifeq ($(shell if [ $(VERSION_CODE) -ge 132608 ]; then echo y; fi),y)
+	cd ${PRL_FREEZE} && make
+endif
+
+install:
+	cd ${PRL_ETH} && make install
+	cd ${PRL_TG} && make install
+	cd ${PRL_FS} && make install
+ifeq ($(shell if [ $(VERSION_CODE) -ge 132608 ]; then echo y; fi),y)
+	cd ${PRL_FREEZE} && make install
+endif
+
+
+cleanclean: clean cleanme
+
+cleanme:
+	cd ${PRL_ETH} && rm -rf .*.cmd .tmp_versions
+	cd ${PRL_TG} && rm -rf .*.cmd .tmp_versions
+	cd ${PRL_FS} && rm -rf .*.cmd .tmp_versions
+	cd ${PRL_FREEZE} && rm -rf .*.cmd .tmp_versions
+
+clean:
+	cd ${PRL_ETH} && make clean
+	cd ${PRL_TG} && make clean
+	cd ${PRL_FS} && make clean
+	cd ${PRL_FREEZE} && make clean
+
diff -rupN src/prl_eth/pvmnet/Makefile.v26 src.new/prl_eth/pvmnet/Makefile.v26
--- src/prl_eth/pvmnet/Makefile.v26	2013-12-03 16:14:32.000000000 +0000
+++ src.new/prl_eth/pvmnet/Makefile.v26	2014-03-06 21:00:13.195839400 +0000
@@ -7,9 +7,19 @@
 obj-m := $(DRIVER).o
 $(DRIVER)-objs := pvmnet.o
 
+MODULE := $(DRIVER)
+MODULE_EXT := .ko
+MODULE_DIR ?= /lib/modules/$(shell uname -r)/extra
+
 all:
 	make -C $(KERNEL_DIR) M=$(PWD)
 
+
+install:
+	@mkdir -p $(MODULE_DIR); \
+	install -m 0664 $(MODULE)$(MODULE_EXT) $(MODULE_DIR); \
+
+
 clean:
 	make -C $(KERNEL_DIR) M=$(PWD) clean
 	rm -f Module*.symvers
diff -rupN src/prl_fs/SharedFolders/Guest/Linux/prl_fs/Makefile.v26 src.new/prl_fs/SharedFolders/Guest/Linux/prl_fs/Makefile.v26
--- src/prl_fs/SharedFolders/Guest/Linux/prl_fs/Makefile.v26	2013-12-03 16:13:45.000000000 +0000
+++ src.new/prl_fs/SharedFolders/Guest/Linux/prl_fs/Makefile.v26	2014-03-06 21:00:13.195839400 +0000
@@ -9,9 +9,19 @@ $(DRIVER)-objs := super.o inode.o file.o
 
 EXTRA_CFLAGS	+= -I$(DRIVER_DIR)/../../../../ -DPRLFS_IGET
 
+MODULE := $(DRIVER)
+MODULE_EXT := .ko
+MODULE_DIR ?= /lib/modules/$(shell uname -r)/extra
+
+
+
 all:
 	make -C $(KERNEL_DIR) M=$(PWD)
 
+install:
+	@mkdir -p $(MODULE_DIR); \
+	install -m 0664 $(MODULE)$(MODULE_EXT) $(MODULE_DIR); \
+
 clean:
 	make -C $(KERNEL_DIR) M=$(PWD) clean
 	rm -f Module*.symvers
diff -rupN src/prl_fs_freeze/Snapshot/Guest/Linux/prl_freeze/Makefile src.new/prl_fs_freeze/Snapshot/Guest/Linux/prl_freeze/Makefile
--- src/prl_fs_freeze/Snapshot/Guest/Linux/prl_freeze/Makefile	2013-12-03 16:14:00.000000000 +0000
+++ src.new/prl_fs_freeze/Snapshot/Guest/Linux/prl_freeze/Makefile	2014-03-06 21:23:13.565303062 +0000
@@ -13,9 +13,23 @@
 #        make KERNEL_DIR=<PATH_TO_KERNEL>
 #
 
+
+DRIVER := prl_fs_freeze
+DRIVER_DIR ?= $(PWD)
+
+obj-m := $(DRIVER).o
+
 KVER ?= $(shell uname -r)
 KERNEL_DIR ?= /lib/modules/$(KVER)/build
 
+MODULE := $(DRIVER)
+MODULE_EXT := .ko
+MODULE_DIR ?= /lib/modules/$(shell uname -r)/extra
+
+
 modules clean:
 	make -C ${KERNEL_DIR} M=${PWD} $@
-obj-m = prl_fs_freeze.o
+
+install:
+	@mkdir -p $(MODULE_DIR); \
+	install -m 0664 $(MODULE)$(MODULE_EXT) $(MODULE_DIR)/
diff -rupN src/prl_tg/Toolgate/Guest/Linux/prl_tg/Makefile.v26 src.new/prl_tg/Toolgate/Guest/Linux/prl_tg/Makefile.v26
--- src/prl_tg/Toolgate/Guest/Linux/prl_tg/Makefile.v26	2013-12-03 16:14:59.000000000 +0000
+++ src.new/prl_tg/Toolgate/Guest/Linux/prl_tg/Makefile.v26	2014-03-06 21:00:13.196839445 +0000
@@ -11,8 +11,13 @@ HFILES = prltg_compat.h ../Interfaces/pr
 obj-m += prl_tg.o
 prl_tg-objs := prltg.o
 
+
+
 KVER ?= $(shell uname -r)
-SRC ?= /lib/modules/$(KVER)/build
+
+MODULE := $(DRIVER)
+MODULE_EXT := .ko
+MODULE_DIR ?= /lib/modules/$(shell uname -r)/extra
 
 _HEADERS_CHECK_DIR := /lib/modules/$(KVER)/source
 ifeq ($(shell if [ -e $(_HEADERS_CHECK_DIR) ]; then echo y; fi),y)
@@ -25,8 +30,17 @@ EXTRA_CFLAGS	+= $(shell \
 	grep -q "\<wait_for_completion_interruptible\>" \
 		$(HEADERS_CHECK_DIR)/include/linux/completion.h || \
 	echo -DPRL_INTERRUPTIBLE_COMPLETION)
+
+all: prl_tg
+
 prl_tg:	$(CFILES) $(HFILES)
-	$(MAKE) -C $(SRC) SUBDIRS=$(shell pwd) SRCROOT=$(shell pwd) modules
+	$(MAKE) -C $(KERNEL_DIR) SUBDIRS=$(shell pwd) SRCROOT=$(shell pwd) modules
+
+
+install:
+	@mkdir -p $(MODULE_DIR); \
+	install -m 0664 $(MODULE)$(MODULE_EXT) $(MODULE_DIR); \
 
 clean:
+	make -C $(KERNEL_DIR) M=$(PWD) clean
 	rm -rf *.o *.ko *.mod* *symvers .tmp_versions .*.cmd prl_tg.ver 
